unrooty/rps_game
# Implement Game logic with UI, JSON response and Tests

## Overview
This PR adds game logic with UI, JSON response and Tests.

## Details
**Game**
* Implement required UI using HTML, Bootstrap 4 and JQuery.
* Implement RockPaperScissorsService to play game.
* Implement RockPaperScissorsController index and play methods.
  - Implement .js.erb rendering for usage from UI.
  - Implement JSON render on JSON requests.
* Add routing and root application route. 
* Add Readme.md setup and play guides.

**Testing**
* Install rspec-rails beta for Rails 6 controller tesing.
* Install webmock to stub the CURB API for unit tests.
* Add unit tests for controller.
* Add unti tests for RPS game service (contains game logic).
* Add integration test to check that service works with CURB API.

## TODO
* Implement Cucumber (with Capybara) tests for UI.
* Improve integration tests.
